app/src/main/java/org/schabi/newpipe/notifications/ChannelUpdates.kt

```diff
@@ -0,0 +1,50 @@
+package org.schabi.newpipe.notifications
+
+import android.content.Context
+import android.content.Intent
+import android.text.TextUtils
+import org.schabi.newpipe.extractor.channel.ChannelInfo
+import org.schabi.newpipe.extractor.stream.StreamInfoItem
+import org.schabi.newpipe.util.NavigationHelper
+
+data class ChannelUpdates(
+    val serviceId: Int,
+    val url: String,
+    val avatarUrl: String,
+    val name: String,
+    val streams: List<StreamInfoItem>
+) {
+
+    val id = url.hashCode()
+
+    val isNotEmpty: Boolean
+        get() = streams.isNotEmpty()
+
+    val size = streams.size
+
+    val text: String
+        get() {
+            val titles = ArrayList<String?>(streams.size)
+            for (stream in streams) {
+                titles.add(stream.name)
+            }
+            return TextUtils.join(", ", titles)
```

Comment:
We've got a small localization problem here, because a comma cannot be used for enumerations in Chinese. But we do not know, if the stream titles are Chinese.
Not sure, if there is a good solution for this.

Reply:
I found solution, hope, it is right